src/Servers/Kestrel/Transport.NamedPipes/src/Internal/NamedPipeConnectionListener.cs

```diff
@@ -26,6 +27,7 @@ internal sealed class NamedPipeConnectionListener : IConnectionListener
     private readonly PipeOptions _inputOptions;
     private readonly PipeOptions _outputOptions;
     private readonly Mutex _mutex;
+    private readonly ConcurrentQueue<NamedPipeServerStream> _streamsCache = new ConcurrentQueue<NamedPipeServerStream>();
```

Comment:
ObjectPool got updated to use CQ. Why not use that here or is the ObjectPool measurable slower?

Reply:
I thought of using ObjectPool but it wasn't referenced.

However! I see that routing uses object pool. And if it's good enough for routing, then it's good enough for Kestrel. And it fixes the issue of pooling more than the upper limit. And I've never used it before and I want to try it out.